bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/persistence/PersistenceResource.java

```diff
@@ -432,7 +432,12 @@ private Response deletePersistenceItemData(@Nullable String serviceId, String it
                     "Persistence service not modifiable: " + serviceId);
         }
 
-        ModifiablePersistenceService mService = (ModifiablePersistenceService) service;
+        try {
+            Item item = itemRegistry.getItem(itemName);
+        } catch (ItemNotFoundException e) {
+            logger.warn("Item not found '{}'.", itemName);
+            return JSONResponse.createErrorResponse(Status.BAD_REQUEST, "Item not found: " + itemName);
+        }
```

Comment:
Is this check really necessary or do persistence services start throwing exceptions when items do not exist in the registry?

With this check it might for instance no longer be possible for users to clean up persisted item state after deleting the items.

Reply:
I think I now see this is only done for storing new data now. :slightly_smiling_face:

Reply:
It's in `deletePersistenceItemData` so my question still stands. :thinking:

Reply:
If we want to allow users to truncate data for not existing items this check indeed is wrong. The question is what happens to the orphaned table then?

I can remove it. If I do I have to change my implementation for JDBC in https://github.com/openhab/openhab-addons/pull/11922 too because it checks for the item to be present in the registry too and returns false for delete operation if not - it does not throw an exception in that case.